Added the details of the movies that the user clicks on.

Kevin-Guillen opened the pull request:
https://user-images.githubusercontent.com/84336422/125792593-be76204d-52f0-4849-9ef4-4718ef872427.mp4

Kevin-Guillen committed:
Added the details of the movies that the user clicks on.

luisjavierglb commented on lib/src/model/movie_result.dart:
```diff
@@ -1,36 +1,25 @@
-  bool? adult;
-  String? overview;
-  String? releaseDate;
+  late bool adult;
```
Whats the purpose of these variables being late instead of nullables ?

Kevin-Guillen replied:
I took a look again and there was no point with this variables being late. I rebuilt this part.

luisjavierglb commented on lib/src/model/movie_result.dart:
```diff
@@ -42,21 +31,21 @@ class MoviesResult {
-      video: parsedJson['video'],
-      title: parsedJson['title'],
-      popularity: parsedJson['popularity'],
+      voteAverage: parsedJson['vote_average'] != null
```
`parsedJson['vote_average'] ?? MovieStrings.defaultVoteAverage` do goo for inline null checks

Kevin-Guillen replied:
done

luisjavierglb commented on lib/src/ui/movies_details_page.dart:
```diff
@@ -0,0 +1,47 @@
+          SliverList(
+            delegate: SliverChildListDelegate(
+              [
+                SizedBox(
```
Remember what we talked about const Widgets

Kevin-Guillen replied:
right , done.

luisjavierglb commented on lib/src/widgets/detailCardsWidgets/custom_appbar.dart:
```diff
@@ -0,0 +1,24 @@
+class CustomAppBar extends StatelessWidget {
+  const CustomAppBar({
+    Key? key,
+    required this.posterPath,
```
Instead of naming this a posterPath lets just name it image right ? so when you use it outside posters it make sense

Kevin-Guillen replied:
i agree , renamed.

luisjavierglb commented on lib/src/widgets/detailCardsWidgets/movie_resume.dart:
```diff
@@ -0,0 +1,28 @@
+import 'package:flutter/material.dart';
+import '../../utils/ui_constants.dart';
+
+class MovieDescription extends StatelessWidget {
```
The name of the file its different form the name of the class, it should be `movie_descritption.dart`

Kevin-Guillen replied:
done.

luisjavierglb reviewed (changes requested):
Some changes requested

FranciscoGaraventa commented on lib/src/ui/movies_details_page.dart:
```diff
@@ -0,0 +1,47 @@
+                ),
+                MovieDescription(overView: data.overView),
+                SizedBox(
+                  height: 10.0,
```
This can be added to a Constant file

Kevin-Guillen replied:
done

FranciscoGaraventa commented on lib/src/widgets/detailCardsWidgets/movie_main_info.dart:
```diff
@@ -0,0 +1,82 @@
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: EdgeInsets.symmetric(horizontal: UiConstants.widgetPadding),
```
You missed a trailling comma here

Kevin-Guillen replied:
added

FranciscoGaraventa commented on lib/src/widgets/detailCardsWidgets/movie_main_info.dart:
```diff
@@ -0,0 +1,82 @@
+                height: UiConstants.posterPathHeight,
+              ),
+            ),
+            tag: id,
```
id can be more descriptive name, you can use heroTag or something like that

Kevin-Guillen replied:
i deleted this hero , i was not using it.

MarcosCGlobant commented on lib/src/ui/movies_details_page.dart:
```diff
@@ -0,0 +1,47 @@
+            delegate: SliverChildListDelegate(
+              [
+                SizedBox(
+                  height: 10.0,
```
remember to move hardcodes to their respective const files

MarcosCGlobant replied:
there is another one on this same file

Kevin-Guillen replied:
done

MarcosCGlobant commented on lib/src/ui/movies_details_page.dart:
```diff
@@ -0,0 +1,47 @@
+                  releaseDate: data.releaseDate,
+                  voteAverage: data.voteAverage,
+                ),
+                MovieDescription(overView: data.overView),
```
missed a trailing comma here

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/widgets/detailCardsWidgets/movie_main_info.dart:
```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+import '../../utils/ui_constants.dart';
+
+class PosterTitle extends StatelessWidget {
```
this class name is different from the file's name, change the file name to keep consistency, and check if this happens in another file

Kevin-Guillen replied:
done

Kevin-Guillen committed:
- Comments fixed:
- rebuilt variables null safety
- rebuild inline null checks with ??
- add "const" to some widgets inside a tree
- renamed posterPath
- renamed two classes
- added a value to the constant file
- removed an unused hero
- Added missing commas
- Some ui changes on details cards

FranciscoGaraventa reviewed (approved):
LGTM

MarcosCGlobant commented on lib/src/model/movies.dart:
```diff
@@ -1,15 +1,9 @@
 import 'movie_result.dart';
 
 class Movies {
-  late final int page;
-  late final int totalResults;
-  late final int totalPages;
```
why did you removed this variables? you should map the entire response, even if you don't use all the elements in it

Kevin-Guillen replied:
re-added

MarcosCGlobant commented on lib/src/ui/movies_details_page.dart:
```diff
@@ -0,0 +1,53 @@
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Container(
+        padding: const EdgeInsets.all(UiConstants.bodyPadding),
```
missed a trailing comma here

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/widgets/detailCardsWidgets/custom_appbar.dart:
```diff
@@ -0,0 +1,24 @@
+    this.movieImage,
+  }) : super(key: key);
+
+  final String? movieImage;
```
why did you make this element nullable?, you have a default value for the image, it shouldn't be null at this point

Kevin-Guillen replied:
got it , fixed.

MarcosCGlobant commented on lib/src/widgets/detailCardsWidgets/custom_appbar.dart:
```diff
@@ -0,0 +1,24 @@
+      expandedHeight: UiConstants.movieDetailsAppBarExpandedHeight,
+      flexibleSpace: FlexibleSpaceBar(
+        background: Image.network(
+          movieImage!,
```
you can remove the "!" operator, after making the parameter non nulable

Kevin-Guillen replied:
removed.

MarcosCGlobant reviewed (approved):
LGTM
